pandas/core/strings.py

```diff
@@ -323,15 +328,23 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
 
     n : int, default -1 (all)
         Number of replacements to make from start
-    case : boolean, default True
-        If True, case sensitive
-    flags : int, default 0 (no flags)
-        re module flags, e.g. re.IGNORECASE
+    case : boolean, optional
+        - if False, case insensitive
+        - Must be None if `pat` is a compiled regex
+    flags : int, optional
+        - re module flags, e.g. re.IGNORECASE
+        - Must be None if `pat` is a compiled regex
```

Comment:
Why was it needed to change the default of `flags` ? Because it is always converted to 0 if not specified (None), so I would just keep it 0